pkg/controllers/work/apply_controller_test.go

```diff
@@ -346,6 +347,46 @@ func TestApplyUnstructured(t *testing.T) {
 	specHashFailObj := correctObj.DeepCopy()
 	specHashFailObj.Object["test"] = math.Inf(1)
 
+	utils.GetObjectFromManifest("../../../test/integration/manifests/resources/test-large-secret.yaml", &largeSecret)
+	largeSecret.ObjectMeta = metav1.ObjectMeta{
+		OwnerReferences: []metav1.OwnerReference{
+			ownerRef,
+		},
+	}
+	rawSecret, _ := json.Marshal(largeSecret)
+	var largeObj unstructured.Unstructured
+	_ = largeObj.UnmarshalJSON(rawSecret)
```

Comment:
please do not ignore the error. you should fail the test if the setup has errors.

https://github.com/golang/go/wiki/CodeReviewComments#handle-errors